src/main/java/org/folio/service/inventory/InventoryItemManager.java

```diff
@@ -142,15 +144,46 @@ public Future<Void> updateItemWithPieceFields(Piece piece, RequestContext reques
     if (piece.getItemId() == null || piece.getPoLineId() == null || piece.getIsBound()) {
       return Future.succeededFuture();
     }
+
+    // Create a promise for this operation
+    Promise<Void> promise = Promise.promise();
```

Comment:
can we change logic in pieces іукмшсу maybe to do sequentual processing(we have utility for this) or some other technic instead having this workaround?

Reply:
Hi @SerhiiNosko you have typo іукмшсу  what does it mean?

Reply:
I mean, is it possible to fix the cause of the race condition instead of addressing the consequences? I don't know where this race condition is occurring, but one potential solution could be to use this chainCall utility method, which will execute features sequentially.
https://github.com/folio-org/mod-orders/blob/master/src/main/java/org/folio/orders/utils/HelperUtils.java#L292

Reply:
thank you Serhii!
 will update the logic, this utility comes handy
the race condition is occurring because of non-blocking nature of vertx 
we should make a blocking operation on get request